drv/stm32xx-sys-api/src/lib.rs

```diff
@@ -52,6 +52,25 @@ pub enum Edge {
     Both = 0b11,
 }
 
+/// Describes which operation is performed by the [`Sys::gpio_irq_control`] IPC.
+#[derive(
+    Copy, Clone, FromPrimitive, PartialEq, Eq, AsBytes, serde::Deserialize,
+)]
+// repr attribute is required for the derived `AsBytes` implementation
+#[repr(u8)]
+pub enum IrqControl {
+    /// Disable any interrupts mapped to the provided notification mask.
+    Disable = 0,
+    /// Enable any interrupts mapped to the provided notification mask.
+    Enable,
+    /// Check if any interrupts mapped to the provided notification mask have
+    /// been triggered, *without* enabling or disabling the interrupt.
+    ///
+    /// If an interrupt is currently enabled, it will remain enabled, while if
+    /// it is currently disabled, it will remain disabled.
+    Check = 0b11,
```

Comment:
The custom discriminator here is surprising; was this deliberate?

Reply:
i...don't remember typing that, actually. it must have been a copy-paste error. i agree that it's not necessary.